OpenRA.Game/Map/CellLayer.cs

```diff
@@ -45,6 +45,19 @@ public void CopyValuesFrom(CellLayer<T> anotherLayer)
 			Array.Copy(anotherLayer.entries, entries, entries.Length);
 		}
 
+		public static CellLayer<T> CreateInstance(Func<MPos, T> initialCellValueFactory, Size size, TileShape tileShape)
+		{
+			var cellInfoLayer = new CellLayer<T>(tileShape, size);
```

Comment:
I'll comment your changes as I go modifying them in order, then I'll commit. This is done